services/frontend/www-app/src/components/BaseMap.vue

```diff
@@ -577,32 +577,100 @@ export default defineComponent({
     });
 
     map.on('load', () => {
-      const layers = map.getStyle().layers;
-      if (layers) {
-        for (const layer of layers) {
-          if (layer.id.startsWith('place_') || layer.id.startsWith('poi_')) {
-            this.on('mouseover', layer.id, (event) => {
-              if (event.features && event.features[0]) {
-                this.setCursor('pointer');
-              } else {
-                console.warn('hovered place without feature', layer, event);
-              }
-            });
-            this.on('mouseout', layer.id, () => {
-              this.setCursor('');
-            });
-            this.on('click', layer.id, (event) => {
-              if (event.features && event.features[0]) {
-                this.touchHandlers
-                  .get('poi_click')
-                  ?.forEach((value) => value(event));
-              } else {
-                console.warn('clicked place without feature', layer, event);
-              }
-            });
-          }
+      const layers: LayerSpecification[] = map.getStyle().layers;
+      if (!layers) {
+        throw new Error('layers must not be empty');
+      }
+
+      for (const layer of layers) {
+        if (layer.id.startsWith('place_') || layer.id.startsWith('poi_')) {
+          this.on('mouseover', layer.id, (event) => {
+            if (event.features && event.features[0]) {
+              this.setCursor('pointer');
+            } else {
+              console.warn('hovered place without feature', layer, event);
+            }
+          });
+          this.on('mouseout', layer.id, () => {
+            this.setCursor('');
+          });
+          this.on('click', layer.id, (event) => {
+            if (event.features && event.features[0]) {
+              this.touchHandlers
+                .get('poi_click')
+                ?.forEach((value) => value(event));
+            } else {
+              console.warn('clicked place without feature', layer, event);
+            }
+          });
         }
       }
+
+      // Add 3-D buildings
+      const render3DZoomLevel = 15;
+      type LerpableValue =
+        | maplibregl.ExpressionSpecification
+        | maplibregl.ColorSpecification
+        | number;
+      function zoomLerp<T>(
+        unzoomedValue: LerpableValue,
+        zoomedValue: LerpableValue
+      ): maplibregl.DataDrivenPropertyValueSpecification<T> {
+        return [
+          'interpolate',
+          ['linear'],
+          ['zoom'],
+          render3DZoomLevel,
+          // There seems to be a bug in maplibre types which forbids passing a value expression like `["get", "render_height"]` here
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          unzoomedValue as any,
+          render3DZoomLevel + 0.5,
+          // There seems to be a bug in maplibre types which forbids passing a value expression like `["get", "render_height"]` here
```

Comment:
I'm going to update our maplibre dependency to see if this problem reproduces on the latest version, and then follow up with them if necessary... updating JS deps can be a bit of a minefield tho.

Reply:
Woohoo, this was fixed in https://github.com/maplibre/maplibre-gl-js/pull/1890 so these hacks can go away once we update to 3.0.0+.  I've added a comment to that effect.